src/internal/utils/global-flags.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+export const awsuiGlobalFlagsSymbol = Symbol.for('awsui-global-flags');
+
+interface GlobalFlags {
+  removeHighContrastHeader?: boolean;
+}
+
+interface ExtendedWindow extends Window {
+  [awsuiGlobalFlagsSymbol]?: GlobalFlags;
+}
+declare const window: ExtendedWindow;
+
+export const getTopWindow = (): ExtendedWindow | undefined => {
+  return window.top as ExtendedWindow;
+};
+
+export const getGlobalFlag = (flagName: keyof GlobalFlags): GlobalFlags[keyof GlobalFlags] | undefined => {
+  try {
+    if (typeof window !== 'undefined') {
+      const topWindow = getTopWindow();
+      return window[awsuiGlobalFlagsSymbol]?.[flagName] ?? topWindow?.[awsuiGlobalFlagsSymbol]?.[flagName];
```

Comment:
I think we should keep `window.top` and `window` separately. Because if something goes wrong in `getTopWindow` , we do not want to bail out immediately.

Something like this:

```js
function readFlag(window) {
  return window[awsuiGlobalFlagsSymbol]?.[flagName]
}

try {
   const ownFlag = readFlag(window);
   if(ownFlag !== undefined) {
       return ownFlag;
   }
   return readFlag(getTopWindow());
}
```

Reply:
Addressed